integreat_cms/static/src/js/pois/opening-hours/component/opening-hours-input-fields.tsx

```diff
@@ -79,9 +79,10 @@ const OpeningHoursInputFields = ({ openingHoursState, selectedDaysState, transla
         // Create new object instead of reference to existing one
         const newOpeningHoursBuffer = deepCopy(openingHoursBuffer);
         newOpeningHoursBuffer.closed = (target as HTMLInputElement).checked;
-        // If closed was checked, disable all day
+        // If closed was checked, disable all day and appointment only
         if (newOpeningHoursBuffer.closed) {
             newOpeningHoursBuffer.allDay = false;
+            newOpeningHoursBuffer.appointmentOnly = false;
```

Comment:
I think in line 99, we need the same statement, right? Otherwise the "appointment" checkbox stays set when "all day" is selected...

Reply:
I don't think so - in the case that the location is open around the clock, you could conceivably still need an appointment. But if it is closed, it's closed, appointment or not.

Reply:
Really? As a user, I would consider these two options contradictory.
Either, I can come without appointment (then I have to come within the correct time slots or at any time if the location is open 24h), or I need an appointment, but then I don't care about the actual opening hours.
To me, it could create misunderstandings if a location is both "open all day" and "open only by appointment"...

Reply:
But what's the difference to e.g. 8:00-17:00, only with appointment? To me those are separate things, one tells me when the place is open, and the other if I need to book an appointment for within that time range.

Setting both appointment only and open all day then just means: Come any time, provided you've called beforehand.

But maybe @osmers can chime in on this? 😄

Reply:
Ah ok, I missed that it's possible to set time slots while "only by appointment" is selected. In my opinion, this shouldn't be possible as well. The service can tell when they have time for an appointment via a third party timeslot booking system or manually via phone or email. Otherwise, I think people might just come during those hours even if they have no appointment. But ok, that's just my perspective, I'm open to input here.

Reply:
I think @charludo is right bcs there are a lot of official places that write opening hours and by appointment only at the same time: 
![image](https://github.com/digitalfabrik/integreat-cms/assets/64901227/78f4eff6-c167-4e8b-9972-4a1d5f10a56b)
I think by appointment only should be selectable when opening times are put in as well as when it's closed all day. And it obviously needs to be selectable on a day to day basis. So Monday should be able to be different from Tuesday and so on.

Reply:
Alright, so no limitations at all then? Including "closed all day, by appointment only"?

Reply:
I mean, it would make sense to let the municipalities decide, right? If that is how opening hours are communicated in their area, why not?

Reply:
OK!

Reply:
Ok, as long as this "appointment only" hint is also shown in the app next to the time slot, I agree that this is the most informative and does not have much potential for misunderstandings :+1:

Reply:
@timobrembeck true - it definitely needs to be displayed next to the specific day
Is there already an app issue for this feature?